plugins/search/src/components/SearchBar/SearchBar.tsx

```diff
@@ -147,12 +161,19 @@ export type SearchBarProps = Partial<SearchBarBaseProps>;
  * @public
  */
 export const SearchBar = ({ onChange, ...props }: SearchBarProps) => {
-  const { term, setTerm } = useSearch();
+  const { term, setTerm, types } = useSearch();
 
   const handleChange = (newValue: string) => {
-    setTerm(newValue);
-    if (onChange) onChange(newValue);
+    if (onChange) {
+      onChange(newValue);
+    } else {
+      setTerm(newValue);
+    }
   };
 
-  return <SearchBarBase value={term} onChange={handleChange} {...props} />;
+  return (
+    <AnalyticsContext attributes={{ types: types.join(',') }}>
```

Comment:
Nice!  Might recommend applying a `sort` on types so that various combinations are more easily groupable?

And maybe namespace the attribute a little?  `types` is could be a lot of things in different contexts, but `searchTypes` is pretty clear.

Might be worth putting this in the `SearchContextProvider` itself, so that it's always applied to an event happening in a search context, regardless of the event.

Reply:
great ideas! updated!